### PR TITLE
Fix build breaks on Windows and fix timezone, DST for all platforms

### PR DESCRIPTION
**Pull request by sanmut**

### CODE REVIEW

File: source/core/Date.cpp [hunk 1]
```diff
@@ -311,14 +372,7 @@ namespace Vireo {
...
 #else
-        // Issue #218 TODO -  FIXME: if utcTime is non-zero, time zone bias should be based on utcTime passed in
-        // (in GMT secs since epoch), not current time.
-        // Possibly use SystemTimeToTzSpecificLocalTime
-        
-        // flipping the sign of the time zone
-        TIME_ZONE_INFORMATION timeZoneInfo;
-        GetTimeZoneInformation(&timeZoneInfo);
-        _systemLocaleTimeZone = -int((timeZoneInfo.Bias) * kSecondsPerMinute);
+        _systemLocaleTimeZone = GetTimeZoneOffsetSeconds(utcTime);
```
Comment (cglzaguilar) [inline source/core/Date.cpp]: Glad you took care of this issue, once the pull request is in, please go ahead and close Issue #218

File: source/core/Date.cpp [hunk 2]
```diff
@@ -180,27 +233,35 @@ namespace Vireo {
...
 // #elif kVireoOS_emscripten variant deleted (03/2017); localtime_r works correctly
 #elif kVireoOS_windows
-        // Issue #218 TODO - FIXME: return time zone info/dst based on timestamp.Integer(), not current time.
-        // Possibly use SystemTimeToTzSpecificLocalTime
         TIME_ZONE_INFORMATION timeZoneInfo;
-        int rc = GetTimeZoneInformation(&timeZoneInfo);
-        wchar_t *timeZoneName = L"TimeZoneUnknown";
-        if (rc == TIME_ZONE_ID_STANDARD)
+        int rc = GetTimeZoneInformationForYear(year, NULL, &timeZoneInfo);
+        char timeZoneName[kTempCStringLength] = "TODO-TimeZone";
```
Comment (spathiwa) [inline source/core/Date.cpp]: "TODO" doesn't seem like a string that should be exposed to the user if an error.

File: source/core/Date.cpp [hunk 3]
```diff
@@ -113,22 +167,21 @@ namespace Vireo {
...
             if (prev == ' ' && input[i] != ' ')
             {
-                output += input[i];
+                output[j] = input[i];
+                ++j;
             }
             prev = input[i];
         }
-        return output;
+        output[j] = 0;
```
Comment (spathiwa) [inline source/core/Date.cpp]: I don't have objections to your C string refactor of this, but is this really the only way to get an abbreviated time zone on Windows? (I think Devin did this originally.)
Did you look for a different API that doesn't require use to hackishly abbreviate the string by looking for spaces?

File: source/core/Date.cpp [hunk 4]
```diff
@@ -53,6 +53,60 @@ namespace Vireo {
 
     Int32 Date::_systemLocaleTimeZone = 0;
 
+#if kVireoOS_windows
+    // UnixTimeToFileTime and UnixTimeToSystemTime are from https ://support.microsoft.com/en-us/help/167296/how-to-convert-a-unix-time-t-to-a-win32-filetime-or-systemtime
+    void UnixTimeToFileTime(time_t t, LPFILETIME pft)
```
Comment (sanmut) [inline source/core/Date.cpp]: Good point. Done.

File: source/core/Date.cpp (hunk 3 above)
Comment (sanmut) [inline source/core/Date.cpp]: No. LV's behavior seems to be locale specific. I haven't spent much time on this. I am not sure whether we want to abbreviate at all just to make the tests pass or make the tests platform specific and go with whatever the default is for the platform (don't abbreviate for Windows and use the abbreviated timezones for Linux/Mac). Let's make a final decision on that in the future, we can spend a bit more effort on this to clean this up.

File: source/core/Date.cpp (hunk 2 above)
Comment (sanmut) [inline source/core/Date.cpp]: Changing to "UnknownTimezone"

File: source/core/Date.cpp (hunk 1 above)
Comment (sanmut) [inline source/core/Date.cpp]: :-) Will close the issues after this gets merged. Thanks.

**Pull request merged**